### PR TITLE
Improved SyncObjectSpawner and fix connectOnAwake bug (@TheYoxy helped)

### PR DESCRIPTION
**Pull request by Upd4ting**
Overview
Improved SyncObjectSpawner by adding more generic method to spawn, search and delete.
Fix a bug when connectOnAwake was to false and was still trying to connect on awake.

Contributors
- TheYoxy
- Upd4ting

### CODE REVIEW

**Comment by FlorianJa** [comment]
Even if you just made changes in the examples, this are breaking changes. Especially the changes in the SyncObjectSpawner.cs

**Comment by Upd4ting** [comment]
Hey!
I will do the changes you requested.
For the SharingStage, the changes are allowing me to start disconnect and then connect. 
Because with the actual system without our changes, when connectOnAwake is to false the SharingStage try to connect to localhost even if connectOnAwake is to false. With theses changes, when connectOnAwake is to false, the SharingStage won't try to connect and we will be able to connect ourself later.

**Comment by Upd4ting** [comment]
Any news?

**Comment by StephenHodgson** [comment]
Sorry been a bit busy the last few days. We'll review this soon.

File: Assets/HoloToolkit-Examples/Sharing/SharingService/Scripts/SyncObjectSpawner.cs [hunk 1]
```diff
@@ -33,42 +33,52 @@ private void Awake()
...
+        /// </summary>
+        public bool SpawnSyncObject(SyncSpawnedObject syncObject, Vector3 position, Quaternion rotation)
         {
-            Vector3 position = Random.onUnitSphere * 2;
-            Quaternion rotation = Random.rotation;
-
-            var spawnedObject = new SyncSpawnedObject();
-
-            spawnManager.Spawn(spawnedObject, position, rotation, spawnParentTransform.gameObject, "SpawnedObject", false);
+            if (!spawnManager.Spawn(syncObject, position, rotation, spawnParentTransform.gameObject, syncObject.Name.Value, false)) {
```
Comment (davidkline-ms) [inline Assets/HoloToolkit-Examples/Sharing/SharingService/Scripts/SyncObjectSpawner.cs]: why not just return the value from Spawn?

ex: return spawnManager.Spawn( … );

File: Assets/HoloToolkit/Sharing/Scripts/SharingStage.cs [hunk 2]
```diff
@@ -233,9 +233,9 @@ protected override void Awake()
             {
                 AutoDiscoverInit();
             }
-            else
+            else if (connectOnAwake)
             {
-                ManagerInit(connectOnAwake);
+                ManagerInit(true);
```
[the remark this answers is not in the document]
Comment (davidkline-ms) [reply Assets/HoloToolkit/Sharing/Scripts/SharingStage.cs]: I'll defer to @StephenHodgson here.

**Comment by Upd4ting** [comment]
Yeah don't know why I didn't do that, that's fixed!

**Comment by Upd4ting** [comment]
Any news?

**Comment by StephenHodgson** [comment]
My biggest concern with the PR: https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/1911#discussion_r179180196

**Comment by Upd4ting** [comment]
But that changes is required to make the Sharing not connect himself at start if we told him to not connect on awake

**Comment by Upd4ting** [comment]
Any news?

**Comment by Upd4ting** [comment]
@StephenHodgson

**Comment by StephenHodgson** [comment]
Hey, sorry, I've been really busy and haven't had a chance to try this out yet.

**Comment by Upd4ting** [comment]
No problem i'll wait 👍

**Comment by SimonDarksideJ** [comment]
@StephenHodgson can you update your review on this item so we can close it down. No update since May 1st.

**Comment by wiwei** [comment]
Closing issues older than 180 days. If this is still an issue, please reactivate with recent information.

**Pull request closed without merging**